### PR TITLE
Remove DynamicKubeletConfig from init and join workflow

### PR DESCRIPTION
**Pull request by yagonobre**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Remove DynamicKubeletConfig from init and join workflow, this also remove the DynamicKubeletConfig feature gate.

```release-note
kubeadm: deprecate the DynamicKubeletConfig feature gate. The functionality is still accessible by using the kubeadm alpha kubelet enable-dynamic command.
```
/sig cluster-lifecycle
/priority important-soon
/assign @timothysc @fabriziopandini @neolit123
@kubernetes/sig-cluster-lifecycle-pr-reviews

### CODE REVIEW

**Comment by neolit123** [comment]
@yagonobre  @fabriziopandini 
i think we should move it to it's new location in the same PR?

**Comment by yagonobre** [comment]
@neolit123 we already have `kubelet config enable-dynamic` on alpha cmd

**Comment by neolit123** [comment]
>  we already have kubelet config enable-dynamic on alpha cmd

missed it.
ok then this PR is fine.

**Review by neolit123 — APPROVED** [review]
LGTM thanks @yagonobre

**Comment by neolit123** [comment]
please use this:
> kubeadm: deprecate the DynamicKubeletConfig feature gate. The functionality is still accessible by using the `kubeadm alpha kubelet enable-dynamic` command.

File: cmd/kubeadm/app/features/features.go
Comment (neolit123) [inline cmd/kubeadm/app/features/features.go]: sorry, just realized that we cannot remove it directly given the gate is BETA already.

we have to fist deprecate it and keep it around for 2 releases given the original feature still exists in the kubelet:
https://github.com/kubernetes/website/blob/master/content/en/docs/reference/using-api/deprecation-policy.md#deprecation

i tried to explain that in the last two meetings.

/hold
/assign @timothysc

**Comment by yagonobre** [comment]
@neolit123 done!

File: cmd/kubeadm/app/features/features.go
Comment (yagonobre) [reply cmd/kubeadm/app/features/features.go]: What we will do? Reopen the phase pr and mark the feature gate as deprecated?

Comment (neolit123) [reply cmd/kubeadm/app/features/features.go]: i will leave to the others for further comments.
@luxas @fabriziopandini

**Review by fabriziopandini — APPROVED** [review]
@yagonobre thanks for this PR!
/approve 

@neolit123 @timothysc my taking from https://github.com/kubernetes/website/blob/master/content/en/docs/reference/using-api/deprecation-policy.md#deprecation is that we can remove the feature but we should keep the feature gate and give a warning for some time.

As per Lucas comment https://github.com/kubernetes/kubernetes/pull/70024#pullrequestreview-173179342, my suggestion is to unblock this PR and tackle the feature gate warning thing once for all flags in the upcoming PR.

I opened https://github.com/kubernetes/kubeadm/issues/1226 to track this issue in the milestone.

**Comment by neolit123** [comment]
/lgtm
tracking of deprecated FGs in: https://github.com/kubernetes/kubeadm/issues/1226

**Pull request merged**
